crates/weaver_emit/src/spans.rs

```diff
@@ -0,0 +1,961 @@
+                        list_of_doubles
+                            .first()
+                            .unwrap_or(&vec![(3.13).into(), (3.15).into()])
+                            .iter()
+                            .map(|d| f64::from(*d))
+                            .collect(),
+                    )),
+                    _ => Value::Array(Array::F64(vec![3.13, 3.15])),
+                },
+                PrimitiveOrArrayTypeSpec::Strings => match &attribute.examples {
+                    Some(Examples::Strings(strings)) => Value::Array(Array::String(
+                        strings.iter().map(|s| s.clone().into()).collect(),
+                    )),
+                    Some(Examples::ListOfStrings(list_of_strings)) => Value::Array(Array::String(
+                        list_of_strings
+                            .first()
+                            .unwrap_or(&vec!["value1".to_owned(), "value2".to_owned()])
+                            .iter()
+                            .map(|s| s.clone().into())
+                            .collect(),
+                    )),
+                    _ => Value::Array(Array::String(vec!["value1".into(), "value2".into()])),
+                },
+            };
+            KeyValue::new(name, value)
+        }
+        AttributeType::Enum { members, .. } => {
+            KeyValue::new(name, Value::String(members[0].value.to_string().into()))
+        }
+        AttributeType::Template(template_type_spec) => {
+            // TODO Support examples when https://github.com/open-telemetry/semantic-conventions/issues/1740 is complete
+            let value = match template_type_spec {
+                TemplateTypeSpec::String => Value::String("template_value".into()),
+                TemplateTypeSpec::Int => Value::I64(42),
+                TemplateTypeSpec::Double => Value::F64(3.13),
+                TemplateTypeSpec::Boolean => Value::Bool(true),
+                TemplateTypeSpec::Strings => Value::Array(Array::String(vec![
+                    "template_value1".into(),
+                    "template_value2".into(),
+                ])),
+                TemplateTypeSpec::Ints => Value::Array(Array::I64(vec![42, 43])),
+                TemplateTypeSpec::Doubles => Value::Array(Array::F64(vec![3.13, 3.15])),
+                TemplateTypeSpec::Booleans => Value::Array(Array::Bool(vec![true, false])),
+            };
+            KeyValue::new(format!("{name}.key"), value)
+        }
+    }
+}
+
+/// Uses the global tracer_provider to emit a single trace for all the defined
+/// spans in the registry
+pub(crate) fn emit_trace_for_registry(registry: &ResolvedRegistry, registry_path: &str) {
+    let tracer = global::tracer("weaver");
+    // Start a parent span here and use this context to create child spans
+    tracer.in_span("weaver.emit", |cx| {
+        let span = cx.span();
+        span.set_attribute(KeyValue::new(
+            "weaver.registry_path",
```

Comment:
This may be a circular problem, but can we register this attribute in SEMCONV?

Either that or use the reserved `otel` namespace